src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -542,7 +542,8 @@ bool isSuitableForZDNN<ONNXReduceMeanOp>(
   }
 
   // Check dimensions.
-  if ((shapeData[2] < 0) || (shapeData[3] < 0) || (shapeData[2] > 1024) ||
+  if ((shapeData[2] == ShapedType::kDynamic) ||
+      (shapeData[3] == ShapedType::kDynamic) || (shapeData[2] > 1024) ||
```

Comment:
should we define some constant for 1024? Not related to this PR.

Reply:
1024 is a HW limit for the NNPA. So you are suggesting that it should be a constant, or a variable that could change in the future for different instances of NNPAs, if that were the case? That make sense.
@tungld if that make sense to you, please create a new PR for Tong's suggestion.